src/util/loopUnrolling.ml

```diff
@@ -244,16 +244,23 @@ let findBreakComparison loopStatement =
     None
 
 let getLoopVar loopStatement func = function
-  | BinOp (op, (Const (CInt (goal, _, _) )), Lval ((Var varinfo), NoOffset), (TInt _)) when isCompare op && not varinfo.vglob ->
+  | BinOp (op, Const (CInt (goal, _, _)), Lval (Var varinfo, NoOffset), TInt _)
+  | BinOp (op, Const (CInt (goal, _, _)), CastE (TInt _, Lval (Var varinfo, NoOffset)), TInt _)
+  | BinOp (op, CastE (TInt _, Const (CInt (goal, _, _))), Lval (Var varinfo, NoOffset), TInt _)
+  | BinOp (op, CastE (TInt _, Const (CInt (goal, _, _))), CastE (TInt _, Lval (Var varinfo, NoOffset)), TInt _) when isCompare op && not varinfo.vglob ->
```

Comment:
This is getting out of hand, so we should refactor this a bit. For example:
1. First match the `BinOp` (with comparison operator and integer type) to just get the two operand expressions.
2. In that case, then apply `Cil.stripCasts` to both of them.
3. In that case, then match on the pair of stripped operand expressions.

([ppx_viewpattern](https://github.com/sim642/ppx_viewpattern/) could simplify this, but I'm not sure if that's production-quality.)